tests/BuildScriptGenerator.Tests/Node/NodeBuildScriptGenerationTest.cs

```diff
@@ -452,6 +453,116 @@ public void GeneratedScript_DoesNotZipNodeModules_IfZipNodeModulesEnvironmentVar
             Assert.True(scriptGenerator.IsCleanRepo(repo));
         }
 
+        [Fact ]
+        public void GeneratedScript_DoesNotConfigureAppInsight_IfAppInsightEnvironmentVariable_NotSet()
+        {
+            var nodeVersion = "10.14.1";
+            // Arrange
+            var scriptGenerator = GetNodePlatformInstance(defaultNodeVersion: nodeVersion);
+            var repo = new MemorySourceRepo();
+            repo.AddFile(PackageJsonWithBuildScript, NodeConstants.PackageJsonFileName);
+            var context = CreateScriptGeneratorContext(repo);
+            context.NodeVersion = nodeVersion;
+            var expected = new NodeBashBuildSnippetProperties(
+                packageInstallCommand: NpmInstallCommand,
+                runBuildCommand: "npm run build",
+                runBuildAzureCommand: "npm run build:azure",
+                hasProductionOnlyDependencies: true,
+                productionOnlyPackageInstallCommand: string.Format(
+                    NodeConstants.ProductionOnlyPackageInstallCommandTemplate,
+                    NpmInstallCommand),
+                compressedNodeModulesFileName: null,
+                compressNodeModulesCommand: null,
+                oryxAppInsightInjectCommand:null);
+
+            // Act
+            var snippet = scriptGenerator.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(snippet);
+            Assert.DoesNotContain("applicationinsights", snippet.BashBuildScriptSnippet);
+            Assert.Equal(
+                TemplateHelpers.Render(TemplateHelpers.TemplateResource.NodeBuildSnippet, expected),
+                snippet.BashBuildScriptSnippet);
+            Assert.True(scriptGenerator.IsCleanRepo(repo));
+        }
+
+        [Fact]
+        public void GeneratedScript_DoesNotConfigureAppInsight_IfNodeVersionCondition_Unsatisfied()
+        {
+            var nodeVersion = "6.9.3";
+            // Arrange
+            var scriptGenerator = GetNodePlatformInstance(defaultNodeVersion: nodeVersion);
+            var repo = new MemorySourceRepo();
+            repo.AddFile(PackageJsonWithBuildScript, NodeConstants.PackageJsonFileName);
+            Environment.SetEnvironmentVariable(Constants.AppInsightKey, "xyz");
```

Comment:
Ideally we do not want unit tests to depend on external source like `System.Environment` here. For this reason we create a layer of indirection through the options class which can be constructed in a test and a value be set on it.
The reason this is bad is because it causes a race condition between tests. Like if a test from a different class(hence would ran in parallel) sets the variable to be empty, then this test can fail (sometimes).